languages/csharp/exercises/concept/user-defined-exceptions/.meta/Example.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+
+public class CalculationException : Exception
+{
+    public CalculationException(int operand1, int operand2, string message, Exception inner) : base(message, inner)
+    // TODO: complete the definition of the constructor
+    {
+        Operand1 = operand1;
+        Operand2 = operand2;
+    }
+
+    public int Operand1 { get; }
+    public int Operand2 { get; }
+}
+
+public class CalculatorTestHarness
+{
+    private Calculator calculator;
+
+    public CalculatorTestHarness(Calculator calculator)
+    {
+        this.calculator = calculator;
+    }
+
+    public string Multiply(int x, int y)
+    {
+        try
+        {
+            TestMultiplication(x, y);
+        }
+        catch (CalculationException cex) when (cex.Operand1 < 0)
+        {
+            return "Multiply failed for a negative value. " + cex.InnerException.Message;
+        }
+        catch (CalculationException cex)
+        {
+            return "Multiply failed for a positive value. " + cex.InnerException.Message;
+        }
+
+        return "Multiply succeeded";
```

Comment:
Maybe we can also return the result of the multiplication in the success case?

Reply:
B) Well, this is testing for overflow conditions, I'm assuming there are other tests for accuracy.  Again the complexity argument although less compelling here.

Reply:
Let's leave it like this. Minor stylistic issue: personally I would put the `return` in the `try` block, to have it be more clear that the `return` belongs to the code running in the `try` block. What do you think?

Reply:
> I would put the return in the try block

This certainly gets the point across to a student better.  I had not really thought about it before but I suppose it makes sense in general.

Reply:
> A bit too marginal to put in the analyzer, perhaps?

I don't mind this in an analyzer actually. This could even be a generic rule.

Reply:
See https://github.com/exercism/csharp-analyzer/issues/55

Reply:
Excellent!